src/dcap_provider.cpp

```diff
@@ -411,7 +435,17 @@ static std::string build_pck_cert_url(const sgx_ql_pck_cert_id_t& pck_cert_id)
     pck_cert_url << '/' << qe_id;
     pck_cert_url << '/' << cpu_svn;
     pck_cert_url << '/' << pce_svn;
-    pck_cert_url << '/' << pce_id << '?';
+    pck_cert_url << '/' << pce_id;
+    if (!eppid.empty())
+    {
+        pck_cert_url << '/' << eppid;

```

Comment:
> eppid [](start = 30, length = 6)

This adds 768 chars to the URL. URL size limit is around 2K. Be careful not to exceed that limit.

Reply:
Alternatively, instead of encoding all this in the URL, possibly put it in the body?

Reply:
Just as a reference - this is how SGX Attestation Service defined [theirs](https://api.portal.trustedservices.intel.com/documentation#pcs-certificate)

https://api.trustedservices.intel.com/sgx/certification/v2/pckcert?encrypted_ppid={}&cpusvn={}&pcesvn={}&pceid={}" -H "Ocp-Apim-Subscription-Key: {subscription key}

Reply:
We can't unify precisely to their API since we can't use the EPPID for identity as far as I know, since it is salted (we use QE ID for this). It's slightly easier to not use name-value pairs to handle cases where a parameter isn't specified. Parsing wise it doesn't matter.

Reply:
I've moved the EPPID into the request body